src/tls/openssl.c

```diff
@@ -529,9 +529,15 @@ static int tls_net_handshake(struct flb_tls *tls,
             pthread_mutex_unlock(&ctx->mutex);
             return -1;
         }
-
-        if (vhost != NULL) {
-            SSL_set_tlsext_host_name(session->ssl, vhost);
+        if (tls->vhost != NULL) {
+            SSL_set_tlsext_host_name(session->ssl, tls->vhost);
+            /* set host name validation only if vhost is configured
+             * explicitely */
+            X509_VERIFY_PARAM *param = SSL_get0_param(session->ssl);
```

Comment:
Move the definition of this variable to the prolog of the function and rename it with a meaningful name.

Reply:
Even though the documentation doesn't state it `SSL_get0_param` can return `NULL` under certain conditions, add code to handle that case gracefully.